### DISCUSSION TITLE
Add tasks with a file

### DISCUSSION THREAD

**Discussion by a-delannoy** [u8slvn/hyperfocus#9] [Ideas]
Hey,

I was thinking about adding a feature like a bulk creation of tasks.

We could give hyperfocus a simple file formatted as a YAML list, for example, to retrieve task names from.

For example, given the following sample file content:
```yaml
- Task1
- This is another task
```
 Hyperfocus would then create two tasks respectively named "Task1" and "This is another task".

This could also be done using an interactive text editor.

What do you think about this ?

**Comment by u8slvn** [comment]
Hey, I already thought about creating multiple tasks at once. IMO, hyperfocus should stay really easy to use and not feature overwhelmed, so I don't like the idea of using files to import tasks but the idea of using the text editor sounds great.

I'd like to propose the following implementation based on your idea:

Add a `bulk` flag. This flag should be used alone, i.e. no `--details`. 

```bash
hyf add -b
# or
hyf add --bulk
```

When used, this flag open the default user's editor and you can just write one task per line (no need to use the YAML syntax):

```bash
My task
Another task
And a last one
```

Of course empty lines will have to be removed and task's titles will have to be striped.
This proposal doesn't handle details option but maybe it could be done like that:

```bash
My task
Another task | This task has details
And a last one
```

And with a simple parsing it should be easy to extract the task title and the details.

What do you think of something like that?

**Comment by a-delannoy** [comment]
Hey,

I like the idea about the `bulk` flag opening the editor.
However, the `|` separator prevent the multiline details and add some work to handle a custom format.

Though I understand your intent of keeping it as simple as possible, this is the reason why I suggested  using YAML. It is rather user friendly, already quite well-known and the parsing is trivial since it is already implemented in python libs.

With a very simple format like: 
```yaml
- name: I am a task
  details: I am the details
- name: I am another task
```
... we would make multiline details possible since YAML has support for multiline strings and keep it rather basic and user friendly.

If you don't like it, I would suggest digging around your idea of "one task per line" which I think is quite simple and nice. However, regarding the details of tasks, I am a bit hesitant with the `|` separator.

**Comment by u8slvn** [comment]
I close the discussion since we talked about it IRL and the first step of the feature is now done.
See: #11
